api-put-object.go

```diff
@@ -63,6 +116,16 @@ func (c Client) PutObject(bucketName, objectName string, data io.Reader, size in
 		return 0, err
 	}
 
+	// get reader size.
+	size, err := getReaderSize(data)
+	if err != nil {
+		return 0, err
+	}
+
+	if size > int64(maxMultipartPutObjectSize) {
+		return 0, ErrInvalidArgument("Input file size is bigger than the supported maximum of 5TiB.")
```

Comment:
ErrInvalidArgument("Object size exceeds maximum limit of 5TiB.")

S3 response is also intuitive. 
EntityTooLarge: Your proposed upload exceeds the maximum allowed object size.